### PR TITLE
Add more observation families to ``kulprit``

### PR DESCRIPTION
**Pull request by yannmclatchie**
## Description

Add the Binomial and Poisson families to ``kulprit``.

## Checklist

- [x] Tests covering the new functionality have been added
- [x] Documentation has been updated OR the change is too minor to be documented
- [ ] Changes are listed in the `CHANGELOG.md` OR changes are insignificant

### CODE REVIEW

File: src/kulprit/projection/projector.py [hunk 1]
```diff
@@ -158,18 +158,23 @@ def project_names(self, term_names: Sequence[str]) -> SubModel:
...
+        # NOTE: I'm not sure if there's a better way to initialize an InferenceData from
+        # a dataset and from a dictionary. Maybe it's worth trying with "add_groups"
+        
         # build idata object for the projected model
-        new_idata = az.data.from_dict(
-            posterior=projected_posterior,
-            observed_data=observed_data,
-        )
+        new_idata = az.InferenceData(posterior=projected_posterior)
+        new_idata.extend(az.data.from_dict(observed_data=observed_data))
```
Comment (tomicapretto) [inline src/kulprit/projection/projector.py:176]: @aloctavodia perhaps you know if there's a cleaner way? `projected_posterior` is a `xarray.Dataset` while `observed_data` is a `dict`.

Comment (aloctavodia) [reply src/kulprit/projection/projector.py:176]: maybe this 

```python
new_idata = az.InferenceData(posterior=projected_posterior,
                          observed_data=az.convert_to_dataset(observed_data))
```

File: src/kulprit/projection/projector.py [hunk 2]
```diff
@@ -163,11 +167,16 @@ def project_names(self, term_names: Sequence[str]) -> SubModel:
...
+
+        print(self.idata.posterior.dims)
+        print(new_idata.posterior.dims)
```
Comment (aloctavodia) [inline src/kulprit/projection/projector.py:179]:
```suggestion
```

**Comment by aloctavodia** [comment]
Actually, I updated the code to run with Bambi from main. I have not updated the version of the dependencies

**Review by aloctavodia — APPROVED** [review]
LGTM!

**Pull request merged**